hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -69,7 +68,7 @@
       LoggerFactory.getLogger(PipelineManagerImpl.class);
 
   // Limit the number of on-going ratis operation to be 1.
-  private final Lock lock;
+  private final ReentrantReadWriteLock lock;
```

Comment:
`PipelineStateManagerV2Impl`  has an internal lock too ,  write options , such as `removePipeline` ,will be protected by this lock.
```
  public void removePipeline(HddsProtos.PipelineID pipelineIDProto)
      throws IOException {
    lock.writeLock().lock();
    try {
      .............
    } finally {
      lock.writeLock().unlock();
    }
  }
```

can we take it into to account?
since `PipelineStateManagerV2Impl` is a private member of `PipelineManagerImpl`,  `PipelineStateManagerV2Impl` can be totally protected by `PipelineManagerImpl`. if we add a lock to `PipelineManagerImpl`, can we remove the internal lock of `PipelineStateManagerV2Impl`?

Reply:
Can we do that in separate jira if we can remove it. I can open a new jira for that.
For this jira, lets stick with fixing problem.

Reply:
> Can we do that in separate jira if we can remove it. I can open a new jira for that.
sure, agree